src/components/global/BaseTextInput.vue

```diff
@@ -2,56 +2,88 @@
 import { defineComponent } from 'vue';
 
 import SearchIcon from '@/assets/icons/search-icon.svg';
+import ViewOffIcon from '@/assets/icons/view-off-icon.svg';
+import ViewOnIcon from '@/assets/icons/view-on-icon.svg';
 
 export default defineComponent({
   name: 'BaseTextInput',
   components: {
-    SearchIcon
+    SearchIcon,
+    ViewOffIcon,
+    ViewOnIcon
   },
   props: {
-    inputName: {
-      type: String,
-      default: 'BaseInput'
-    },
-    inputType: {
+    type: {
       type: String,
       default: 'text'
     },
     modelValue: {
       type: String,
       default: ''
     },
-    placeholder: {
-      type: String,
-      default: 'placeholder'
-    },
     isSearch: {
       type: Boolean,
       default: false
     }
   },
+  data() {
+    return {
+      isPasswordVisible: false,
+      inputType: this.type
+    };
+  },
+  methods: {
+    toggleIsPasswordVisible() {
+      this.isPasswordVisible = !this.isPasswordVisible;
+      this.inputType === 'text'
+        ? (this.inputType = 'password')
+        : (this.inputType = 'text');
+    }
```

Comment:
So this is where I think this component does too much 😄
Don't get me wrong, it works just fine and is done very well!
But! Does a `BaseTextInput` is always used for passwords and has this toggle?

This is a perfect job for a new component (say, `PasswordInput`) that deals with passwords only and perhaps cleverly utilizes some slots to display those icons to toggle it 😄

If we introduce such slots, this way our `BaseTextInput` component could also be a base for other inputs that have some sort of buttons/toggles inside them for various purposes 💡 

And also as a final note - accessibility 😄 - we should utilize a `button` element to toggle the password input on/off (the icons could be inside that `button` element).
We could strip it off from all the button styles (leaving only the focus; remember the rule - don't break working defaults 😄) and we get a nicely accessible toggle both from keyboard and mouse 👍

Reply:
Ok, so in order to keep things simple I overcomplicated the input in the end..:sweat_smile:
Definitely some great advice here, I'll try to divide this code into two better & more accessible components!

Reply:
No worries!
This is actually the good reason why we create these "BaseSomething" components 😄 - they can be great base to build upon some more complex, specific components 😄